### PR TITLE
Add `ToGlueRow` derive macro and `values_from` insert builder

### LINKED ISSUE #1849: ToGlueRow would help
As `FromGlueRow` helps with `SELECT (row → struct)`, there should be `ToGlueRow` for helping with `INSERT (struct → values).` If we want to avoid SQL and insert with AST builder only, we need to do this : 

```rust
fn build_insert_items_stmt(batch: &[Item]) -> gluesql_core::ast::Statement {
    let rows = batch.iter().map(|it| {
        vec![
            num(it.id),
            text(&it.name),
            num(it.price_cents),
        ]
    });

    insert_into("items")
        .columns(vec!["id", "name", "price_cents"])
        .values(rows)
        .build()
        .expect("build insert")
}
```

There should be support for converting our struct values to rows without dealing with field types.

### PR DESCRIPTION
**Pull request by juhee200**
Define `ToGlueRow` to derive column metadata and row conversion from structs. Add `values_from` to support struct-based insertion in the query builder, replacing column expressions. Update error handling to enforce non-empty `values_from` inputs. Expand tests to validate struct-based insertion via `values_from`. Document struct-based insertion support and field renaming.

Closes #1849 

## Summary by CodeRabbit

- **New Features**
  - Added a `ToGlueRow` derive to convert Rust structs into insertable row values, including renamed columns.
  - Added `values_from` to insert multiple rows from structs with automatic column/value alignment.
  - Re-exported the `ToGlueRow` derive from the Rust package.

- **Bug Fixes**
  - `values_from` now returns a clear error when given no rows, and when row field counts don’t match.

- **Documentation**
  - Added an “Insert from Structs” section with examples.

- **Tests**
  - Added compile-fail and runtime coverage, including invalid derive targets and duplicate column detection.

### CODE REVIEW

**Comment by panarch** [comment]
Separate from the functional review, now that this crate has both `FromGlueRow` and `ToGlueRow`, how about moving the new `ToGlueRow` implementation into a submodule? We can leave extracting the existing `FromGlueRow` implementation for a follow-up issue to keep this PR scoped.

File: core/src/query_builder/insert.rs [hunk 1]
```diff
@@ -38,6 +39,31 @@ impl InsertNode {
...
+                    .collect::<Vec<_>>()
+                    .into()
+            })
+            .collect();
+
+        Ok(InsertSourceNode {
+            insert_node: self.columns(columns),
+            source: QueryNode::Values(values),
+        })
+    }
```
[the remark this answers is not in the document]
Comment (juhee200) [reply core/src/query_builder/insert.rs:75]: The derive-generated impl always matches, but since ToGlueRow can be hand-implemented, added a per-row arity check that returns a dedicated ValuesFromColumnCountMismatch { expected, found } error instead of silently building a malformed INSERT.

**Comment by juhee200** [comment]
> Separate from the functional review, now that this crate has both `FromGlueRow` and `ToGlueRow`, how about moving the new `ToGlueRow` implementation into a submodule? We can leave extracting the existing `FromGlueRow` implementation for a follow-up issue to keep this PR scoped.

That sounds like a good plan. I just pushed the changes to move it into a submodule. 
Does this look like the direction you were thinking of?

**Comment by panarch** [comment]
> That sounds like a good plan. I just pushed the changes to move it into a submodule.
> Does this look like the direction you were thinking of?

Yes, this is the direction I had in mind. Keeping the ToGlueRow expansion and its tests in a dedicated module looks good, while leaving FromGlueRow unchanged keeps this PR focused. Thanks!

**Comment by panarch** [comment]
Could we reject duplicate final column names during derive expansion?

For example:

```rust
#[derive(ToGlueRow)]
struct Item {
    value: i64,

    #[glue(rename = "value")]
    previous_value: i64,
}

let item = Item {
    value: 10,
    previous_value: 20,
};
```

This generates column metadata and values equivalent to:

```rust
columns = ["value", "value"];
values = [10, 20];
```

The new arity check does not catch this because both lengths are `2`. If the target table has another nullable or defaulted column, the insert can still succeed: the executor finds the first `"value"` entry, inserts `10`, and the later value `20` is never used.

Could we detect duplicate final column names with a `HashSet` and return a derive-time error? A compile-fail test covering a field-name/rename collision would also help prevent regressions.

**Comment by juhee200** [comment]
> Could we reject duplicate final column names during derive expansion?
> 
> For example:
> 
> ```rust
> #[derive(ToGlueRow)]
> struct Item {
>     value: i64,
> 
>     #[glue(rename = "value")]
>     previous_value: i64,
> }
> 
> let item = Item {
>     value: 10,
>     previous_value: 20,
> };
> ```
> 
> This generates column metadata and values equivalent to:
> 
> ```rust
> columns = ["value", "value"];
> values = [10, 20];
> ```
> 
> The new arity check does not catch this because both lengths are `2`. If the target table has another nullable or defaulted column, the insert can still succeed: the executor finds the first `"value"` entry, inserts `10`, and the later value `20` is never used.
> 
> Could we detect duplicate final column names with a `HashSet` and return a derive-time error? A compile-fail test covering a field-name/rename collision would also help prevent regressions.


Oh, I hadn't thought of that edge case — great catch, thank you! The macro now emits a compile-time error (duplicate column name \value`) when a rename collides with another column, with the error span pointing at the offending #[glue(rename)]` attribute. Added a unit test and a trybuild compile-fail case to cover it.

File: macros/src/to_glue_row.rs [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+            #gluesql_crate::translate::IntoParamLiteral::into_param_literal(
+                ::core::clone::Clone::clone(&self.#field_ident)
+            )
+        });
+    }
+
+    let columns_len = column_names.len();
+
+    let expanded = quote! {
+        impl #gluesql_crate::row_conversion::ToGlueRow for #ident {
```
[the remark this answers is not in the document]
Comment (juhee200) [reply macros/src/to_glue_row.rs:71]: Good point — the generated impl indeed doesn't handle generics. That said, the existing FromGlueRow derive has the same limitation, and split_for_impl alone wouldn't fully solve it either: a type parameter used as a field would still need IntoParamLiteral + Clone bounds to compile. Rather than making the two derives inconsistent, I'd prefer to leave this as is for now and handle generics support for both ToGlueRow and FromGlueRow together in a follow-up issue, alongside the module extraction already planned. Happy to file the issue.

**Review by panarch — APPROVED** [review]
Looks all nice!
I registered the following issue - https://github.com/gluesql/gluesql/issues/1951

Thanks a lot for the contribution! 👍 👍 👍
